### PR TITLE
HSEARCH-4198 Transactional outbox event creation with Spring and JTA

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-4198

### CODE REVIEW

File: mapper/orm/src/main/java/org/hibernate/search/mapper/orm/outbox/impl/OutboxTableSendingPlan.java [hunk 1]
```diff
@@ -49,18 +50,43 @@ public void discard() {
...
+					.autoClose( false )
+					.connectionHandlingMode( PhysicalConnectionHandlingMode.DELAYED_ACQUISITION_AND_RELEASE_AFTER_TRANSACTION )
+					.openSession();
+			return sendAndReportOnSession( temporarySession, entityReferenceFactory );
+		}
+		finally {
+			if ( temporarySession != null ) {
+				temporarySession.close();
+			}
+		}
```
Comment (yrodiere) [inline mapper/orm/src/main/java/org/hibernate/search/mapper/orm/outbox/impl/OutboxTableSendingPlan.java:73]: A `try-with-resources` block would be nice here. And that's not just nitpicking:

* with try-with-resources, if `sendAndReportOnSession` throws an exception and `temporarySession.close()` also throws an exception, the second exception will correctly be added as suppressed to the first one.
* with `try/finally`, in the same situation, the first exception will be ignored and only the second one (from `temporarySession.close()`) will be propagated.

File: mapper/orm/src/main/java/org/hibernate/search/mapper/orm/outbox/impl/OutboxEventBackgroundExecutor.java [hunk 2]
```diff
@@ -104,6 +104,7 @@ public HibernateOrmOutboxWorker(SessionFactoryImplementor sessionFactory) {
 					List<OutboxEvent> outboxes = finder.findOutboxEvents( session, batchSize );
 					if ( outboxes.isEmpty() ) {
 						// Nothing to do, try again later (complete() will be called, re-scheduling the polling for later)
+						transactionHelper.commit( session );
```
Comment (yrodiere) [inline mapper/orm/src/main/java/org/hibernate/search/mapper/orm/outbox/impl/OutboxEventBackgroundExecutor.java:107]: Nice catch.

File: .gitignore [hunk 3]
```diff
@@ -45,3 +45,7 @@ nbactions.xml
 .DS_Store
 .attach_pid*
 
+# Bitronix transaction logs
+part1.btm
+part2.btm
```
Comment (yrodiere) [inline .gitignore:50]: There may be a `part3` from time to time? Better safe than sorry:

```suggestion
*.btm
```

**Review by yrodiere — COMMENTED** [review]
Looks good apart from a minor comment. Feel free to merge after it's addressed. Thanks!

**Comment by fax4ever** [comment]
Full build: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-search/detail/PR-2574/3/pipeline

**Comment by fax4ever** [comment]
Thanks @yrodiere, I applied the suggestions.
I'll merge it as soon as the full-CI passes.

**Comment by fax4ever** [comment]
Merged, thanks!

**Pull request merged**